crates/ibc/src/applications/transfer/context.rs

```diff
@@ -51,6 +54,60 @@ pub trait Ics20Reader: ChannelReader {
     }
 }
 
+pub trait Ics20ChannelKeeper {
+    fn store_send_packet_result(&mut self, result: SendPacketResult) -> Result<(), Ics04Error> {
+        self.store_next_sequence_send(
+            result.port_id.clone(),
+            result.channel_id.clone(),
+            result.seq_number,
+        )?;
+
+        self.store_packet_commitment(
+            result.port_id,
+            result.channel_id,
+            result.seq,
+            result.commitment,
+        )?;
+        Ok(())
+    }
+
+    fn store_packet_commitment(
+        &mut self,
+        port_id: PortId,
+        channel_id: ChannelId,
+        sequence: Sequence,
+        commitment: PacketCommitment,
+    ) -> Result<(), Ics04Error>;
+
+    fn store_next_sequence_send(
+        &mut self,
+        port_id: PortId,
+        channel_id: ChannelId,
+        seq: Sequence,
+    ) -> Result<(), Ics04Error>;
+}
+
+impl<T: ChannelKeeper> Ics20ChannelKeeper for T {
```

Comment:
Can we use the `where` notation? I find it clearer when I skim the codebase

Reply:
https://github.com/cosmos/ibc-rs/pull/244/commits/b5dc9f36943ade62cf4891f02e01b7301d6283eb